setup.py

```diff
@@ -43,50 +36,82 @@ def build_extension(self, ext):
-                              libraries=['gssapi_krb5'],
-                              define_macros=defines)
+    defines = [
+        ("HAVE_GSS_SET_CRED_OPTION", hasattr(libgssapi_krb5, "gss_set_cred_option")),
+        (
+            "HAVE_GSSSPI_SET_CRED_OPTION",
+            hasattr(libgssapi_krb5, "gssspi_set_cred_option"),
+        ),
+    ]
+    lw_krb_module = Extension(
+        "pike.kerberos",
+        [
+            "pykerb/base64.c",
+            "pykerb/kerberosbasic.c",
+            "pykerb/kerberos.c",
+            "pykerb/kerberosgss.c",
+            "pykerb/kerberospw.c",
+        ],
+        libraries=["gssapi_krb5"],
+        define_macros=defines,
+    )
 except OSError:
     try_krb = False
 
+
 def pike_suite():
-    return unittest.defaultTestLoader.discover('pike/test', pattern='*.py')
+    return unittest.defaultTestLoader.discover("pike/test", pattern="*.py")
+
 
 def run_setup(with_extensions):
     ext_modules = []
-    cmdclass = { "build_py": ve_build_py }
+    cmdclass = {"build_py": ve_build_py}
     if with_extensions:
         ext_modules.append(lw_krb_module)
         cmdclass = dict(cmdclass, build_ext=ve_build_ext)
-    setup(name='Pike',
-          version=__version__,
-          description='Pure python SMB client',
-          author='Brian Koropoff',
-          author_email='Brian.Koropoff@emc.com',
-          url='https://github.com/emc-isilon/pike',
-          packages=['pike', 'pike.test'],
-          install_requires=['pycryptodomex'],
-          ext_modules=ext_modules,
-          test_suite='setup.pike_suite',
-          cmdclass=cmdclass,
-          )
+    setup(
+        name="pike-smb2",
+        use_scm_version=True,
+        setup_requires=["setuptools_scm"],
+        description="Pure python SMB client",
+        author="Brian Koropoff",
+        author_email="Brian.Koropoff@emc.com",
+        maintainer="Masen Furer",
+        maintainer_email="Masen.Furer@dell.com",
+        url="https://github.com/emc-isilon/pike",
+        packages=["pike", "pike.test"],
+        python_requires=">=2.7,!=3.0.*,!=3.1.*,!=3.2.*,!=3.3.*,!=3.4.*,<4",
+        install_requires=["pycryptodomex", "future"],
+        ext_modules=ext_modules,
+        test_suite="setup.pike_suite",
+        cmdclass=cmdclass,
+        classifiers=[
+            "Development Status :: 4 - Beta",
+            "Intended Audience :: Developers",
+            "Topic :: Software Development :: Testing",
+            "Programming Language :: Python",
+            "Programming Language :: Python :: 2",
+            "Programming Language :: Python :: 2.7",
+            "Programming Language :: Python :: 3",
+            "Programming Language :: Python :: 3.5",
+            "Programming Language :: Python :: 3.6",
+            "Programming Language :: Python :: 3.7",
+            "Programming Language :: Python :: Implementation :: CPython",
+            "Operating System :: OS Independent",
+        ],
```

Comment:
I recommend adding

    classifiers=[
    ...
    "Environment :: Console",
     "License :: Other/Proprietary License",
     ...
    ]
    keywords='smb smb-testing smb-client',

I'd update the _Manage Topics_ on the pike github landing page to match the `keywords`.

https://pypi.org/classifiers/